src/odemis/acq/stream/_static.py

```diff
@@ -389,22 +400,27 @@ def __init__(self, name, data, *args, **kwargs):
                         sempos_cur = sempos
                         break
                 self._pos[sempos_cur + (d.metadata.get(MD_POL_MODE, None),)] = img.ensure2DImage(d)
+
+                # if polarimetry info available, collect all pol pos data for each ebeam pos
+                if model.MD_POL_MODE in d.metadata.keys():
+                    if sempos_cur not in self._pos_polarimetry.keys():
+                        self._pos_polarimetry[sempos_cur] = {d.metadata[model.MD_POL_MODE]: img.ensure2DImage(d)}
+                    else:
+                        self._pos_polarimetry[sempos_cur][d.metadata[model.MD_POL_MODE]] = img.ensure2DImage(d)
+
                 sempositions.add(sempos_cur)
                 if MD_POL_MODE in d.metadata:
                     polpositions.add(d.metadata.get(MD_POL_MODE))
+
             except KeyError:
                 logging.info("Skipping DataArray without known position")
 
-        # Cached conversion of the CCD image to polar representation
-        # TODO: automatically fill it in a background thread
-        self._polar = {}  # dict tuple (float, float, str or None) -> DataArray
-
         # SEM position VA
         # SEM position displayed, (None, None) == no point selected (x, y)
         self.point = model.VAEnumerated((None, None),
-                     choices=frozenset([(None, None)] + list(sempositions)))
+                                        choices=frozenset([(None, None)] + list(sempositions)))
 
-        if self._pos:
+        if self._pos and len(sempositions) > 0:
```

Comment:
How could `self._pos` be not empty, but `sempositions` be empty? IOW, what is the purpose of this extra check?

Reply:
There was some issue I remember, but might be that this issue is no longer valid or I rewrote the code in between...removed it.